Memory optimization

Jakob8 opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/LucasBrazi06/ev-simulator?pullRequest=6) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LucasBrazi06/ev-simulator?pullRequest=6) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LucasBrazi06/ev-simulator?pullRequest=6) it.</sub>

jerome-benoit commented on src/utils/Configuration.ts:
```diff
@@ -45,6 +45,14 @@ export default class Configuration {
+    return Configuration.objectHasOwnProperty(Configuration.getConfig(), 'chargingStationsPerWorker') ? Configuration.getConfig().chargingStationsPerWorker : 1;
+  }
+
+  static getChargingStationIdSuffix(): string {
```
That tunable must be put at the charging station template level.

Jakob8 replied:
Really?
I used the suffix to append '-Roaming' to all of the charging station ids
That makes it easier to identify charging stations from one specific test.
I use the suffix to filter the charging stations that I want to set public / private
or to delete charging stations after a load test (because the simulator can create thousands of charging stations that are not longer needed in the tenant)

jerome-benoit replied:
> Really?
> I used the suffix to append '-Roaming' to all of the charging station ids
> That makes it easier to identify charging stations from one specific test.
> I use the suffix to filter the charging stations that I want to set public / private
> or to delete charging stations after a load test (because the simulator can create thousands of charging stations that are not longer needed in the tenant)

Really ;) The charging station naming is defined at the template level. The global configuration is only here for transversal tunables like the kind of processes policy, the logging, etc.

jerome-benoit reviewed (approved):
I'll merge it and then polish the code.

Jakob8 commented:
A few remarks to my changes:
General
•	I added the authorize request to authorize the tagID on the backend 
•	I added the config to set a charging station ID suffix 
•	The instance index is now part of the CS-ID 
o	Only if more than 1 instance are running 
Performance
•	I added the config to set the number of charging stations per worker 
o	= 1 (default) -> every CS runs in an own worker, like before 
o	> 1 -> not all charging stations run in one single worker. A new worker is created when number of CSs per worker is reached 
o	= very large number -> all charging station run in a single worker
o	I think this option is better than either running all CSs in individual worker or running all in the main thread 
•	Running multiple CSs in a worker saves a lot of memory 
o	each worker takes up about 8MB memory -> 100 workers need 800-900 MB
o	100 charging stations in 1 worker -> 50MB
•	the workers start sequentially now 
o	that leads to massive memory saving on start time 
o	downside: a take longer until all CSs are running 
Scripts
•	I added a script to set specific CSs public or private
o	e.g. use CS-ID suffix to filter the right CSs 
•	I added a script delete specific CSs 
o	e.g. use CS-ID suffix to filter the right CSs
